### PR TITLE
Rename test data folder from mock_data to test_files

### LINKED ISSUE #585: Rename test data folder from mock_data to test_data or similar
Noticed (#576) we use folder name `mock_problems` for holding test files we using in tests, this is slightly confusing since it indicate it has to do with mocking (i.e. using python mock) which is not the case.

Here change this to `test_data` (or test_problems or test_files for example)

### PR DESCRIPTION
**Pull request by Shanrahan16**
#### Description of Work

Fixes #585

#### Testing Instructions

1.
2.
3.

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?

### CODE REVIEW

**Comment by Shanrahan16** [comment]
I think it's failing the two tests due to a mantid-related issue
image

**Review by AndrewLister-STFC — CHANGES_REQUESTED** [review]
Looks brill, there's one minor place that still has "mock_problems". Then we can merge

File: fitbenchmarking/utils/tests/test_fitbm_result.py [hunk 1]
```diff
@@ -30,7 +30,7 @@ def setUp(self):
         Setting up FitBenchmarking results object
         """
         self.options = Options()
-        mock_problems_dir = os.path.dirname(inspect.getfile(mock_problems))
+        mock_problems_dir = os.path.dirname(inspect.getfile(test_files))
         problem_dir = os.path.join(mock_problems_dir, "cubic.dat")
```
[the remark this answers is not in the document]
Comment (Shanrahan16) [reply fitbenchmarking/utils/tests/test_fitbm_result.py:34]: Ah yes, I looked at this and wasn't sure whether to change the directory name as well or not, but it makes sense if they both match

**Review by AndrewLister-STFC — DISMISSED** [review]

File: docs/source/extending/parsers.rst [hunk 2]
```diff
@@ -87,15 +87,15 @@ To add a new fitting problem definition type, complete the following steps:
...
      This will be used to verify that the problem can be run by scipy, and that
      accuracy results do not change unexpectedly in future updates.
      If the software used for the new parser is pip-installable, and the
      installation is done via FitBenchmarking's ``setup.py``, then add the
-     same example to ``fitbenchmarking/mock_problems/default_parsers/``.
+     same example to ``fitbenchmarking/test_files/default_parsers_set/``.
 
      As part of this, the ``systests/expected_results/all_parsers.txt`` file,
-      and if necessary the ``systests/expected_results/default_parsers.txt`` file,
+      and if necessary the ``systests/expected_results/default_parsers_set.txt`` file,
```
Comment (AndrewLister-STFC) [inline docs/source/extending/parsers.rst:98]: Are these now csv? @Shanrahan16

Comment (Shanrahan16) [reply docs/source/extending/parsers.rst]: Yes, sorry, I updated it in the other one I think, so I thought it had picked them all up as conflicts- I've changed it now though, and checked there are no more like this :)

File: docs/source/extending/parsers.rst [hunk 3]
```diff
@@ -87,15 +87,15 @@ To add a new fitting problem definition type, complete the following steps:
...
-     ``fitbenchmarking/mock_problems/all_parser_set/``.
+     ``fitbenchmarking/test_files/all_parser_set/``.
      This will be used to verify that the problem can be run by scipy, and that
      accuracy results do not change unexpectedly in future updates.
      If the software used for the new parser is pip-installable, and the
      installation is done via FitBenchmarking's ``setup.py``, then add the
-     same example to ``fitbenchmarking/mock_problems/default_parsers/``.
+     same example to ``fitbenchmarking/test_files/default_parsers_set/``.
 
      As part of this, the ``systests/expected_results/all_parsers.txt`` file,
```
Comment (AndrewLister-STFC) [inline docs/source/extending/parsers.rst:97]:
```suggestion
     As part of this, the ``systests/expected_results/all_parsers.csv`` file,
```

**Review by AndrewLister-STFC — APPROVED** [review]

**Pull request merged**